src/transformers/generation_utils.py

```diff
@@ -644,34 +588,15 @@ def _generate_beam_search(
                 )
 
             scores = F.log_softmax(next_token_logits, dim=-1)  # (batch_size * num_beams, vocab_size)
-
-            scores = self.postprocess_next_token_scores(
-                scores=scores,
-                input_ids=input_ids,
-                no_repeat_ngram_size=no_repeat_ngram_size,
-                bad_words_ids=bad_words_ids,
-                cur_len=cur_len,
-                min_length=min_length,
-                max_length=max_length,
-                eos_token_id=eos_token_id,
-                repetition_penalty=repetition_penalty,
-                batch_size=batch_size,
-                num_beams=num_beams,
-            )
+            if sampler:
```

Comment:
I think here we will not be able to keep backwards probability with `beam_search` + `sampling` because `top_k_top_p_filtering` is applied after the beam scores are added. I think from a logical point of view it does make more sense to apply `top_k_top_p_filtering` after adding the beam scores. On the other hand beam search sampling is not used that much and definitely an edge case....

Reply:
IIUC the proposal would be, get the raw logits, normalize, add beam scores and then perform sampling using the transformed distribution? That makes sense to me and I'll make the change; it seems like a design decision as to how to make these probability shifts interact with beam search. Is it covered in any literature?

Reply:
Yeah, I would think the distribution should be transformed after the  beam scores have been added. I don't know any literature here though. I'm not too concerned about beam search + sampling, but I'm not sure if we also restrict "greedy" beam search this way for future use cases. @yjernite @srush - do you have more insight here maybe?